src/xpcc/architecture/interface/memory.hpp

```diff
@@ -0,0 +1,109 @@
+// coding: utf-8
+/* Copyright (c) 2016, Roboterclub Aachen e.V.
+ * All Rights Reserved.
+ *
+ * The file is part of the xpcc library and is released under the 3-clause BSD
+ * license. See the file `LICENSE` for the full license governing this code.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef XPCC_MEMORY_HPP
+#define XPCC_MEMORY_HPP
+
+#include <stdint.h>
+#include <xpcc/architecture/utils.hpp>
+#include <xpcc/architecture/interface/register.hpp>
+
+/**
+ * @ingroup		interface
+ * @defgroup	memory_traits	Memory traits
+ *
+ * Traits of dynamic memory.
+ *
+ * @see		driver
+ * @author	Niklas Hauser
+ */
+
+namespace xpcc
+{
+
+/// Describes access and type of dynamic memory.
+/// @ingroup memory_traits
+enum class
+MemoryTrait : uint16_t
+{
+	AccessSBus = Bit0,			///< Memory is accessible via System-Bus
```

Comment:
These definitions need to be available in an internal C-header for use with in C-sources for libc.
At the moment multiple files need to be synchronized:
- this header,
- 3 heap implementations (TLSF, newlib and block allocator)
- 4 linkerscripts for stm32